### PR TITLE
Feature/more fine grained permissions

### LINKED ISSUE #259: More fine grained permissions with verbose names
Now that certain permissions are starting to require more and more components, certain names no longer properly reflect the elements they affect. Permissions such as can_view_assignment_participants does not just concern viewing participants, but also their journal, their files, and probably more. In order to preserve integrity for permissions, we should rename problematic permissions to more descriptive counterparts.

The offending permissions will need to be identified, and a better name for them should be chosen on a case-by-case basis.

### PR DESCRIPTION
**Pull request by mjvkeulen**
## Description
Implements the new permission scheme back and front. Still requires #201 to ensure backend holds to it.

image

## Summary of changes
- Fixes #259

### CODE REVIEW

File: src/django/VLE/views/role.py [hunk 1]
```diff
@@ -189,8 +194,14 @@ def partial_update(self, request, pk):
                 role = factory.make_role_default_no_perms(new_role['name'], course)
             serializer = RoleSerializer(role, data=new_role, partial=True)
             if not serializer.is_valid():
+                print('serializer is not valid')
```
Comment (Larspolo) [inline src/django/VLE/views/role.py]: remove print

**Comment by engelhamer** [comment]
You could include creation of the file itself in the make setup sometime.

**Comment by mjvkeulen** [comment]
> You could include creation of the file itself in the make setup sometime.

A dummy file is now added, I recommend simply not tracking it locally after changes.
git update-index --skip-worktree src/django/VLE/settings/email.py

https://git-scm.com/docs/git-update-index

File: src/django/VLE/models.py [hunk 2]
```diff
@@ -176,29 +177,52 @@ class Role(models.Model):
...
+
+    def save(self, *args, **kwargs):
+        if self.can_add_course_users and not self.can_view_course_users:
+            raise ValidationError('A user needs to view course users in order to add them.')
+
+        if self.can_delete_course_users and not self.can_view_course_users:
+            raise ValidationError('A user needs to view course users in order to remove them.')
+
+        if self.can_edit_course_user_group and not self.can_view_course_users:
+            raise ValidationError('A user needs to view course users in to manage user groups.')
```
[the remark this answers is not in the document]
Comment (mjvkeulen) [reply src/django/VLE/models.py]: Good catch

**Pull request merged**